test/parallel/test-whatwg-url-searchparams.js

```diff
@@ -44,6 +44,18 @@ n = 0;
 for (val of sp.values()) {
   assert.strictEqual(val, String(values[n++]));
 }
+n = 0;
+sp.forEach(function(val, key, obj) {
+  assert.strictEqual(this, undefined);
+  assert.strictEqual(key, 'a');
+  assert.strictEqual(val, String(values[n++]));
+  assert.strictEqual(obj, sp);
+});
+sp.forEach(function() {
+  assert.strictEqual(this, m);
+}, m);
+assert.throws(() => sp.forEach(), TypeError);
```

Comment:
Instead of the `TypeError` constructor, can you pass a regular expression like `/^TypeError: The `callback` argument must be a function$/`